include/swift/AST/DiagnosticsParse.def

```diff
@@ -131,17 +131,17 @@ ERROR(lex_unicode_escape_braces,none,
 ERROR(lex_illegal_multiline_string_start,none,
       "multi-line string literal content must begin on a new line", ())
 ERROR(lex_illegal_multiline_string_end,none,
-      "multi-line string literal content must not be on the delimiter's line", ())
-ERROR(lex_insufficient_string_indent,none,
-      "string literal content is insufficiently indented", ())
-ERROR(lex_unexpected_tab_in_string_indent,none,
-      "string literal content is indented with a tab where a space is expected", ())
-ERROR(lex_unexpected_space_in_string_indent,none,
-      "string literal content is indented with a space where a tab is expected", ())
+      "multi-line string literal closing delimiter must begin on a new line", ())
+ERROR(lex_inconsistent_string_indent,none,
+      "line does not start with same indentation as closing delimiter of multi-line string literal", ())
+NOTE(note_insufficient_string_indent,none,
+      "unexpectedly found non-indentation character here", ())
+NOTE(note_unexpected_char_in_string_indent,none,
+      "unexpected %select{space|tab}0 in indentation here", (bool))
+NOTE(note_matching_indentation_char_here,none,
+      "should match %select{space|tab}0 in same position before closing delimiter", (bool))
```

Comment:
Firstly, it's helpful to flag both locations because either one could be mistaken. But the main reason I do this is to illustrate the connection between the delimiter indentation and the line's indentation. I think this is important for people who are new to the multiline string feature and don't understand how its indentation behavior works yet.

It'd be nice to have a single note which pointed to both locations and said "these have to match", but our diagnostics don't allow that, so that leaves us with two separate notes. One of them could be merged with the error message, but both need to exist.

We could, perhaps, do this, though:

```
test/Parse/multiline_errors.swift:46:3: error: character does not match indentation before closing delimiter of multi-line string literal
        Pi
  ^^^^^^
test/Parse/multiline_errors.swift:47:3: note: tab there doesn't match corresponding space here
    """ // expected-error@-1{{line does not start with same indentation as closing delimiter of multi-line string literal}}
  ^
test/Parse/multiline_errors.swift:46:1: note: change indentation of this line to match closing delimiter
        Pi
^~~~~~~~
         
```

That would carry the same pieces of information in fewer lines